### PR TITLE
Deprecation of MATLAB table as input to addRow and addColumn DynamicTable methods

### PR DESCRIPTION
**Pull request by cechava**
calling `addTableRow` or `addTableColumn` functions will throw an error instructing users to use key-value pairs instead

### CODE REVIEW

File: +tests/+sanity/GenerationTest.m [hunk 1]
```diff
@@ -38,18 +38,21 @@ function dynamicTableMethodsTest(testCase)
                 'colnames', colnames);
             
             id = primes(2000) .';
-            for i = 1:100
+            for i = 1:200
```
Comment (bendichter) [inline +tests/+sanity/GenerationTest.m:41]: why generate so many random variables? It seems like this test would work just as well with 5 rows.

**Review by ln-vidrio — APPROVED** [review]

**Comment by cechava** [comment]
@bendichter if this also looks good to you. As requested, I reduced the number of random numbers generated for the tests

File: +tests/+sanity/GenerationTest.m [hunk 2]
```diff
@@ -37,21 +37,24 @@ function dynamicTableMethodsTest(testCase)
                 'description', 'test dynamic table column',...
                 'colnames', colnames);
             
-            id = primes(2000) .';
-            for i = 1:100
+            id = primes(200) .';
```
Comment (bendichter) [inline +tests/+sanity/GenerationTest.m:40]: why are you generating primes though?

Comment (bendichter) [reply +tests/+sanity/GenerationTest.m:40]: again, you are using `.'` in your syntax and I do not understand why. This appears to do the same thing as `primes(200)`'

Comment (cechava) [reply +tests/+sanity/GenerationTest.m:40]: Commented on this on PR #378 as well. Copying here:
This was there before and doesn't hinder any test so I thought I would keep it. From what I gather, prime numbers are used as ids to have a very different way to enumerate rows to make sure the the 'useID' argument of getRow works properly. We could also do something simpler like enumerating rows in intervals of 5, if you prefer.

Comment (cechava) [reply +tests/+sanity/GenerationTest.m:40]: I can remove the dot. As you point out, it does nothing in this context.

Comment (ln-vidrio) [reply +tests/+sanity/GenerationTest.m:40]: So I've been using the `.'` transpose style. It usually does nothing but I wanted to be clear on what I'm doing to the matrix itself. The difference between `transpose` and `ctranspose` here:
https://www.mathworks.com/help/matlab/ref/transpose.html
https://www.mathworks.com/help/matlab/ref/ctranspose.html

Comment (ln-vidrio) [reply +tests/+sanity/GenerationTest.m:40]: The usage of `primes` was also me and it's just as @cechava says I just needed a non-linear id vector. As long as it's not the default `id` vector structure it should be testing the same thing. That's just there to differentiate row index and row IDs.

Comment (bendichter) [reply +tests/+sanity/GenerationTest.m:40]: @ln-vidrio thanks for pointing me to that reference. I was unaware of the difference between transpose and ctranspose. I don't think we really deal with complex values in NWB, but I can see why you'd default to `.'` nonetheless. Looks good to me!

**Review by bendichter — APPROVED** [review]

File: +tests/+sanity/GenerationTest.m (hunk 2 above)
Comment (cechava) [reply +tests/+sanity/GenerationTest.m:40]: @ln-vidrio yes, for pointing to that.

**Pull request merged**